doc/tensor.md

```diff
@@ -1327,36 +1331,66 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
 ```
 
 <a name="torch.Tensor.gather"/>
-### [Tensor] gather(src, dim, index) ###
+### [Tensor] gather(dim, index) ###
```

Comment:
> essentially src:func(src), i.e. it's an in-place operation

Sorry to nitpick, but I think it is much more common for `src:func(args)`
to return a freshly allocated tensor that holds the result, rather than a
destination for the result being explicitly provided like in:
`dst:func(src, args)`.

On Fri, Jul 3, 2015 at 5:21 PM Dominik Grewe <notifications@github.com>
wrote:

> In doc/tensor.md
> <https://github.com/torch/torch7/pull/274#discussion_r33873037>:
>
> > @@ -1327,36 +1331,66 @@ x:indexFill(2,torch.LongTensor{4,2},-10)
> >  ```
> >
> >  <a name="torch.Tensor.gather"/>
> > -### [Tensor] gather(src, dim, index) ###
> > +### [Tensor] gather(dim, index) ###
>
> It's actually quite common to be able to call a function both as
> src:func() and as dst:func(src). However, often the first case means
> essentially src:func(src), i.e. it's an in-place operation. This is where
> gather differs.
>
> I'd say the most common uses of gather will be dst:gather(src, dim, idx),
> because you want to avoid memory allocations where possible. So maybe
> that's an argument for leaving src in the header :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/torch/torch7/pull/274/files#r33873037>.
>